project3/participant/tasks/multicast_receiver_task.cpp

```diff
@@ -0,0 +1,46 @@
+#include "multicast_receiver_task.h"
+
+#include <loguru.hpp>
+
+namespace participant_tasks {
+MulticastReceiver::MulticastReceiver(
+    const std::shared_ptr<ConsoleTask>& console_task, std::string log_location,
+    int port)
+    : port_(port), log_location_(log_location), console_task_(console_task) {}
+
+thread_pool::Task::Status MulticastReceiver::SetUp() {
+  struct sockaddr_in address = participant_util::MakeAddress(port_);
+  messenger_fd_ = participant_util::SetUpListenerSocket(address);
+  loguru::add_file(log_location_.c_str(), loguru::FileMode::Truncate,
```

Comment:
I wasn't sure what log level to log here. I want this log to **only** log messages received by this task.

Reply:
It was more to satisfy part of the project description. Namely, the part where participants are supposed to log messages sent over through that group while they are registered.

Reply:
Ah, that's my fault for not reading the description. After reading it, I think that this is probably a decent way to go. However, as it stands, I think this will also capture log messages from every other thread. We might have to log messages manually to mitigate that.

Reply:
I should be more specific. By "log messages" I meant multicast messages sent through the group.